plugins/modules/nmcli.py

```diff
@@ -2604,7 +2636,23 @@ def main():
                               parent=dict(type='str', required=True),
                               promiscuous=dict(type='bool'),
                               tap=dict(type='bool'))),
-            wireguard=dict(type='dict'),
+            wireguard=dict(
+                type='dict',
+                options=dict(
+                    fwmark=dict(type='int', required=False),
+                    ip4_auto_default_route=dict(type='bool', required=False),
+                    ip6_auto_default_route=dict(type='bool', required=False),
+                    listen_port=dict(type='int', required=False),
+                    mtu=dict(type='int', required=False),
+                    peer_routes=dict(type='bool', required=False),
+                    private_key=dict(type='str', required=False, no_log=True),
+                    private_key_flags=dict(type='int', required=False, no_log=True),
+                    peer_public_key=dict(type='str', required=False),
+                    peer_allowed_ips=dict(type='str', required=False),
+                    peer_endpoint=dict(type='str', required=False),
+                    peer_persistent_keepalive=dict(type='int', required=False),           
```

Comment:
Adding `options` is a breaking change and has no chance of getting merged.

Before the user was free to supply whatever they want under `wirdguard`. With your change only the options written here are accepted, and nothing else.

(Also even if this would be ok: you are not using the correct option names here. `peer_persistent_keepalive` should be `peer-persistent-keepalive`, for example.)